src/internal_rep.rs

```diff
@@ -370,6 +370,8 @@ impl TypeInfo {
     // All types must inherit from some built in.  Get one for this type.
     // It's possible to inherit from multiple built-ins, so order matters here.  We return the
     // first type in order of preference.
+    // Using find gets us a &&str, and dereferencing inserts an unnecessary copy
+    #[allow(clippy::manual_find)]
```

Comment:
`inter()` and `into_iter()` on a slice return the same thing, which is an issue to follow this tips. However, we can do `constants::BUILT_IN_TYPES.iter().find(|t| self.is_type_by_name(types, t)).copied()`, which will only copy a pointer/reference, so I think it does the same as the current code.

Reply:
Thanks for the suggestion.  Switching to that seems to work.